app/controllers/api/v1/subject_sets_controller.rb

```diff
@@ -86,7 +86,6 @@ def destroy_relation(resource, relation, value)
   private
 
   def remove_linked_set_member_subjects(resource, set_member_subjects)
-    SubjectWorkflowCount.where(set_member_subject: set_member_subjects).delete_all
```

Comment:
I just put in a `dependent: :destroy` on the has_many relations, and set the foreign keys to restrict so we can't accidentally make a mess on the console. Probably the best of both explicitness and referential integrity.